src/lib.rs

```diff
@@ -1251,15 +1251,24 @@ pub enum Expression {
 
 pub use block::Block;
 
+/// The value of a case for a switch statement.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serialize", derive(Serialize))]
+#[cfg_attr(feature = "deserialize", derive(Deserialize))]
+pub enum SwitchCaseValue {
+    Sint(i32),
+    Uint(u32),
+}
+
 /// A case for a switch statement.
 // Clone is used only for error reporting and is not intended for end users
 #[derive(Clone, Debug)]
 #[cfg_attr(feature = "serialize", derive(Serialize))]
 #[cfg_attr(feature = "deserialize", derive(Deserialize))]
 pub struct SwitchCase {
     /// Value, upon which the case is considered true.
-    pub value: i32,
-    /// Body of the cae.
+    pub value: SwitchCaseValue,
```

Comment:
is this really needed?
  1. different cases inside the same switch must have the same type
  2. both i32 and u32 can be losslessly converted to each other

So it seems to me that we could just keep `i32` (or `u32`) in here, and just interpret it differently based on some bit inside the switch itself.

Reply:
My fear is how you would check that they have the same type?

Reply:
WGSL frontend can check this. The IR would just assume they are the same type

Reply:
And the backends would need to check the type of the selector and write the postfix as needed?

Reply:
correct, yes